src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -287,6 +308,66 @@ public TestResult findCorrespondingResult(String id) {
         return Collections.emptyList();
     }
 
+    /**
+     * Start for Test Result Trends.
+     * From here onwards start the code area responsible for generating various test result trends.
+     */
+
+    public hudson.tasks.junit.TestResult loadXml(){return null;}
+
+    /**
+     * A method for getting the list of packages for all levels of hierarchy.
+     * @return Array of packages of all hierarchies.
+     */
+    public String[] getProjectList(){
+
+        /*
+         * If project list is already ready no need to construct again.
+         */
+        if(projectList!=null) return projectList;
+        hudson.tasks.junit.TestResult r = loadXml();
+        Collection<SuiteResult> suiteList = r.getSuites();
+
+        /*
+         * A set for the package names.
+         */
+        Set<String> projectSet = new HashSet<String>();
+        for(SuiteResult sr: suiteList){
+            for(CaseResult cr: sr.getCases()){
```

Comment:
Try to avoid shortcuts, you can use suiteResult and caseResult instead of sr and cr

Reply:
I did that because initial developers also used these names only for same purpose but I'll make the changes and make them more verbose. Thanks.

Reply:
@sanjeet006py If there is an existing code badly written, we need not follow the same. We can keep our changes clean.